### PR TITLE
Grammar check - "a [a,e]" rule.

### PR DESCRIPTION
**Pull request by bchavez**
This PR covers **A through E** for the following grammar rule:

**Grammar Rule:**
https://owl.english.purdue.edu/owl/resource/591/01/
```
"A" goes before words that begin with consonants.
a cat
a dog
a purple onion
a buffalo
a big apple

"An" goes before words that begin with vowels:
an apricot
an egg
an Indian
an orbit
an uprising
```

**NOTES:**
* Some `*.resx` files were touched.
* Covers XML docs
* Covers source code comments too

**STATUS:**
:red_circle:  *In review. Don't merge yet!*

----

Holy cow :cow:, there are tons of these everywhere. Thinking I should start an OSS business doing this. haha :rofl: j/k

Thanks,
:mag_right: ***Linterman***

:zap: :runner: ***["I was struck by lighting. Walkin' down the street..."](https://www.youtube.com/watch?v=iypUpv9xelg)***

### CODE REVIEW

**Comment by danmosemsft** [comment]
@dotnet-bot test linux arm release build (myget flaky)

File: src/System.Private.Xml/src/System/Xml/Schema/XdrBuilder.cs [hunk 1]
```diff
@@ -1120,7 +1120,7 @@ private static void XDR_EndAttributeType(XdrBuilder builder)
                     goto cleanup;
                 }
 
-                // a attributes of type id is not supposed to have a default value
+                // an attributes of type id is not supposed to have a default value
```
Comment (JonHanna) [inline src/System.Private.Xml/src/System/Xml/Schema/XdrBuilder.cs]: *An attribute*

**Comment by EgorBo** [comment]
@bchavez what tool are you using? 🙂

**Comment by JonHanna** [comment]
There's a few cases of *a XML* which breaks this rule (since its the sound rather than the letter that matters, and /ɛks'ɛm'ɛl/ begins with a vowel, so it should be *an XML*. Likewise *a HTTP* (fine for an en-IE speaker like me where that's pronounced /heɪtʃ'tiː'tiː'piː/ but most dialects have it as /eɪtʃ'tiː'tiː'piː/ or /eɪtʃ'ti'ti'pi/ so it should be *an HTTP*). There may be other cases of acronyms where the initial sound does not match the "AEIOU" way of identifying a vowel. Conversely there's a few cases of *a Uri* (I know some people pronounce *URL* as a homophone of *earl*, but does anyone do similar with *URI*, as in that case either would be acceptable).

Similarly, there's a few places with *an Unicode* which should be *a Unicode* (/ˈjuni'koʊd/ or /ˈjuːni'kəʊd/). You might be interested in tracking those down too.

**Pull request merged**